app/views/components/index/ChangeContactSection.scala.html

```diff
@@ -0,0 +1,33 @@
+@*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import components._
+@this(link: Link)
+
+@(isBusiness: Boolean, businessName: String, changeContactDetailsUrl: String)(implicit messages: Messages)
+
+    @changeContactLink = @{if(isBusiness){
+        link(changeContactDetailsUrl, messages("organisationIndex.p3", businessName), Some("change-contact-details"))
```

Comment:
In the scenario where `isBusiness=true` and `sub.businessName.getOrElse("")` (`IndexController.scala` on line 56) returns an empty string then the rendered message would be incorrect as below:
`Change the contact details for `
instead of 
`organisationIndex.p3 = Change the contact details for <some-biz-name>`

Would a default value other than an empty string be safer (this may be a design question)?

Reply:
Hm, i suppose we need to know is that's something we need to protect against; a case where we have a business without a name -
perhaps a Problem page in that case would be best? then have businessName as Option[String]=None